tapgarden/custodian.go

```diff
@@ -582,6 +587,18 @@ func (c *Custodian) mapToTapAddr(walletTx *lndclient.Transaction,
 		return nil, fmt.Errorf("unable to encode address: %w", err)
 	}
 
+	// Skip already completed events.
+	ctxt, cancel = c.CtxBlocking()
+	existingEvent, err := c.cfg.AddrBook.QueryEvent(ctxt, addr, op)
+	cancel()
+	if err == nil && existingEvent.Status >= address.StatusCompleted {
```

Comment:
I think we end up dropping the err here? We could safely ignore the `ErrNoRows` equivalent though.